db/migrate/20200914163852_add_dttp_id_to_trainees.rb

```diff
@@ -0,0 +1,7 @@
+class AddDttpIdToTrainees < ActiveRecord::Migration[6.0]
+  def change
+    # TODO: make this null: false
+    add_column :trainees, :dttp_id, :uuid
```

Comment:
```suggestion
    add_column :trainees, :dttp_id, :uuid, :default => "uuid_generate_v4()"
```

Then in the model there will be no need for `before_create` as it's generated by database by default.

Reply:
I tried that first but that function might not be enabled on all postgres-ses (It's not on mine for example). I'm also a bit hesitant about making that a database responsibility as it makes the process a bit more opaque when you're reading the code.